File: models/boss.js
import { Ally } from './classes.js'

class Boss extends Ally {
  constructor(name, text, health, attack, thwart) {
    super(name, text, health, attack, thwart)
  }

  getDamage = function(enemyDeck) {
    var card = enemyDeck.pop()
    var boost = card.attack
    var damage = this.attack + boost
    return damage
  }
  getThreat = function(enemyDeck) {
    var card = enemyDeck.pop()
    var boost = card.thwart
    var threat = this.thwart + boost
    return threat
  }

  /* Villain Phases */
  threatAndDamage = function(hero, scheme, villainDeck) {
    if (hero.inHeroForm = false) {
      scheme.threat = scheme.threat + this.getThreat
    } else {
      //do damage to hero
    }
  }

  damageOnly = function(hero, scheme, villainDeck) {
    hero.villainPhase(this.attack)
  }

  villainPhase = function(hero, scheme, villainDeck) {
    this.damageOnly(hero, scheme, villainDeck)
  }
}

export { Boss }

File: python/models/boss.py
from models.killable import Killable
from config import config, log

class Boss(Killable):
  def __init__(self, name, text, health, attack, thwart):
      super().__init__(name, text, health, attack, thwart)

  def getDamage(self, enemyDeck):
      card = enemyDeck.pop()
      boost = card.attack
      damage = self.attack + boost
      return damage

  def getThreat(self, enemyDeck):
      card = enemyDeck.pop()
      boost = card.thwart
      threat = self.thwart + boost
      return threat

  def threatAndDamage(self, hero, scheme, villainDeck):
      if hero.inHeroForm == False:
          scheme.threat = scheme.threat + self.getThreat(villainDeck)
      else:
          #do damage to hero
          log('do damage to hero')


